dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/ZipkinOtelCollectorTest.java

```diff
@@ -0,0 +1,207 @@
+
+    @ClassRule
+    public static ZipkinContainer zipkinContainer = new ZipkinContainer()
+    .withLogConsumer(new SimpleLogConsumer(ZipkinTest.class, "zipkin"))
+    .withNetwork(network)
+    .withNetworkAliases("zipkin-all-in-one");
+
+    
+    //Copies otel-collector-config.yaml for OpenTelemetry Collector container
+    @ClassRule
+    public static GenericContainer<?> otelCollectorContainer = new GenericContainer<>(
+                    new ImageFromDockerfile().withDockerfileFromBuilder(builder -> builder.from(
+                                                                        ImageNameSubstitutor.instance()
+                                                                        .apply(DockerImageName.parse("otel/opentelemetry-collector:0.74.0"))
+                                                                        .asCanonicalNameString())
+                                                                        .copy("/etc/otel-collector-config.yaml", "/etc/otel-collector-config.yaml")
+                                                                        .build())
+                                    .withFileFromFile("/etc/otel-collector-config.yaml", new File("lib/LibertyFATTestFiles/otel-collector-config.yaml"), 0644))
+                                    .withExposedPorts(OTLP_GRPC_PORT)
+                                    .withCommand("--config=/etc/otel-collector-config.yaml")
+                                    .withNetwork(network)
+                                    .withLogConsumer(new SimpleLogConsumer(JaegerBaseTest.class, "otelCol"));
+
+    public ZipkinQueryClient client = new ZipkinQueryClient(zipkinContainer);
+
+    @Server("spanTestServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        server.addEnvVar(TestConstants.ENV_OTEL_TRACES_EXPORTER, "otlp");
+        server.addEnvVar(TestConstants.ENV_OTEL_EXPORTER_OTLP_ENDPOINT, "http://" + otelCollectorContainer.getHost() + ":" + otelCollectorContainer.getMappedPort(OTLP_GRPC_PORT));
+        server.addEnvVar(TestConstants.ENV_OTEL_SERVICE_NAME, "Test service");
+        server.addEnvVar(TestConstants.ENV_OTEL_BSP_SCHEDULE_DELAY, "100"); // Wait no more than 100ms to send traces to the server
+        server.addEnvVar(TestConstants.ENV_OTEL_SDK_DISABLED, "false"); //Enable tracing
+
+        // Construct the test application
+        WebArchive spanTest = ShrinkWrap.create(WebArchive.class, "spanTest.war")
+                                        .addClass(TestResource.class);
+        ShrinkHelper.exportAppToServer(server, spanTest, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer();
+    }
+
+    @Test
+    public void testBasic() throws Exception {
+        HttpRequest request = new HttpRequest(server, "/spanTest");
+        
+        Thread.sleep(5000);
```

Comment:
You mustn't put sleeps like this in FAT tests.

This whole test class will take at least 25 seconds, just from the sleeps whereas it should usually take about 5 seconds.